### PR TITLE
make the next release, with more features

### PR DESCRIPTION
**Pull request by vtjnash**

### CODE REVIEW

**Comment by IanButterworth** [comment]
Worth a CI rerun?

File: test/runtests.jl [hunk 1]
```diff
@@ -219,29 +226,40 @@ end
...
+    close(lockf2)
 end
-            
+
+@assert !ispath("pidfile")
 @testset "mkpidlock" begin
-    @info "mkpidlock"
     lockf = mkpidlock("pidfile")
     waittask = @async begin
+        @show lockf
```
Comment (vtjnash) [inline test/runtests.jl:256]:
```suggestion
```

File: src/Pidfile.jl [hunk 2]
```diff
@@ -221,12 +224,41 @@ function open_exclusive(path::String;
...
+                rmpath = string(rmpath, isempty(rmpath) ? "" : path_separator,
+                    "\$", _rand_filename(), rmname, ".deleted")
+                ispath(rmpath) || break
             end
+            rename(path, rmpath)
         end
+        rm(rmpath)
+        true
+    catch ex
+        @show ex
```
Comment (vtjnash) [inline src/Pidfile.jl:259]:
```suggestion
```

**Comment by vtjnash** [comment]
looks like this is pretty much ready to go tomorrow, then just need to update the gh-pages branch after merging

**Pull request merged**